haystacker-server/src/main/kotlin/net/sr89/haystacker/server/handlers/GetBackgroundTaskProgressHandler.kt

```diff
@@ -16,10 +16,17 @@ class GetBackgroundTaskProgressHandler(private val taskManager: BackgroundTaskMa
     override fun invoke(request: Request): Response {
         val tid: String = taskId(request)
 
-        val status = taskManager.status(TaskId(UUID.fromString(tid)))
-        val resp = BackgroundTaskStatusResponse(tid, status.state.name, status.description)
+        try {
+            val status = taskManager.status(TaskId(UUID.fromString(tid)))
```

Comment:
I'd create a function such as `fun parseTaskId(): TaskId` which throws `InvalidTaskIdException` (to be created), and then handle that exception in `ExceptionHandlingFilter`.
This way we avoid making this method significantly longer and having to deal with try-catch here

Reply:
Better yet, I'd prefer to have `TaskId.fromString(...)` instead of `fun parseTaskId(taskId: String): TaskId`, so that it's easily discoverable/re-usable

Reply:
Okay sounds good. I will make those changes and revert

Reply:
I have made the recommended changes in new commit